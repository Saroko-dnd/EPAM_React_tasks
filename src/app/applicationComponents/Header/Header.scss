.site-header-navigation {
  a {
    font-size: 2rem;
  }
}

.site-header {
  header {
    font-size: 3rem;
    vertical-align: middle;
  }
}
