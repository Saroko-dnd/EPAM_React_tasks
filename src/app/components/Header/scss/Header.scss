@import "media";
@import "variables";

.site-header-navigation {
  background-color: #1e2228;

  a {
    font-size: 3rem;
  }
}

.site-header {
  width: 100%;

  h1 {
    vertical-align: middle;
  }
}

.nav-menu-button {
  font-size: 4rem;
  color: white;
  background-color: black;
  border: 0;
  padding: 0;
  cursor: pointer;
  transition: all 0.5s;

  &:hover {
    color: $color-menu-button-hover;
  }
}
