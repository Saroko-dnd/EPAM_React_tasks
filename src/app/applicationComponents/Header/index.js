import React from 'react';
import { NavLink } from 'react-router-dom';

import NewsIcon from '../../reusableComponents/svg/NewsIcon';

import './Header.scss';

const Header = () => (
    <header className="site-header d-flex align-items-center flex-column">
        <header className="d-flex align-items-center">
            <span>
                <NewsIcon height="150" width="150" />
            </span>
            News from all over the world
        </header>
        <nav className="site-header-navigation nav nav-pills d-flex justify-content-around">
            <NavLink
                className="nav-item nav-link"
                to="/"
                exact
                // activeStyle={{ color: 'white' }}
            >
                Top news
            </NavLink>
            <NavLink
                className="nav-item nav-link"
                to="/first"
                exact
                // activeStyle={{ color: 'white' }}
            >
                Ultra news
            </NavLink>
            <NavLink
                className="nav-item nav-link"
                to="/second"
                exact
                // activeStyle={{ color: 'white' }}
            >
                Super news
            </NavLink>
            <NavLink
                className="nav-item nav-link"
                to="/third"
                exact
                // activeStyle={{ color: 'white' }}
            >
                Mega news
            </NavLink>
        </nav>
    </header>
);

export default Header;
