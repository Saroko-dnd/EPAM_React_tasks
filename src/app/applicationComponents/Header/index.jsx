import React from 'react';
import { NavLink } from 'react-router-dom';

import NewsIcon from '../../reusableComponents/svg/NewsIcon';

import './Header.scss';

const Header = () => (
    <header className="site-header d-flex align-items-center flex-column">
        <header className="d-flex align-items-center">
            <span>
                <NewsIcon height={150} width={150} />
            </span>
            News from all over the world
        </header>
        <nav className="site-header-navigation nav nav-pills d-flex justify-content-around">
            <NavLink className="nav-item nav-link" to="/" exact>
                Top news
            </NavLink>
            <NavLink className="nav-item nav-link" to="/first" exact>
                Ultra news
            </NavLink>
            <NavLink className="nav-item nav-link" to="/second" exact>
                Super news
            </NavLink>
            <NavLink className="nav-item nav-link" to="/third" exact>
                Mega news
            </NavLink>
        </nav>
    </header>
);

export default Header;
