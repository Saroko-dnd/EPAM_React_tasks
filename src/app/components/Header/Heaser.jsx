import React from 'react';
import PropTypes from 'prop-types';
import { NewsIcon } from 'react-svg-icons-by-igor-saroko';

import NavigationPanel from '../NavigationPanel';
import { navigation } from '../../constants';

import './scss/Header.scss';

class Header extends React.Component {
    shouldComponentUpdate() {
        console.log('header shouldComponentUpdate');
        return true;
    }

    render() {
        return (
            <header className="site-header d-flex align-items-center flex-column">
                <header className="d-flex align-items-center">
                    <span>
                        <NewsIcon
                            height={150}
                            width={150}
                            className="site-header-icon"
                        />
                    </span>
                    <h1>News from all over the world</h1>
                    <button
                        className="nav-menu-button d-block d-sm-none"
                        onClick={() => {
                            this.props.toggleNavLinks(!this.props.navLinksToggled);
                        }}
                    >
                        <span className="fa fa-bars" />
                    </button>
                </header>
                <NavigationPanel
                    className={`site-header-navigation d-sm-flex ${
                        this.props.navLinksToggled ? 'd-flex' : 'd-none'
                    }`}
                    navLinksCallback={this.props.toggleNavLinks}
                    navigationLinks={[
                        navigation.homePage,
                        navigation.newsDetails,
                        navigation.secondPage,
                        navigation.thirdPage,
                    ]}
                />
            </header>
        );
    }
}

Header.propTypes = {
    /* location: PropTypes.shape({ pathname: PropTypes.string.isRequired })
        .isRequired, */
    toggleNavLinks: PropTypes.func.isRequired,
    navLinksToggled: PropTypes.bool.isRequired,
};

export default Header;
