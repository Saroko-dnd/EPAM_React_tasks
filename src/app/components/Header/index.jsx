import React from 'react';
import { NewsIcon } from 'react-svg-icons-by-igor-saroko';

import NavigationPanel from '../NavigationPanel';
import { navigation } from '../../constants';

import './scss/Header.scss';

const Header = () => (
    <header className="site-header d-flex align-items-center flex-column">
        <header className="d-flex align-items-center">
            <span>
                <NewsIcon height={150} width={150} />
            </span>
            News from all over the world
        </header>
        <NavigationPanel
            className="site-header-navigation"
            navigationLinks={[
                navigation.homePage,
                navigation.firstPage,
                navigation.secondPage,
                navigation.thirdPage,
            ]}
        />
    </header>
);

export default Header;
