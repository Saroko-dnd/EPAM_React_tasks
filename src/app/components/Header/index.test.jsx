import Header from './index';

describe('render an Header', () => {
    it('should render Header correctly', () => {
        const wrapper = shallow(<Header navMenuIsOpened={false} toggleNavLinks={() => {}} />);
        expect(wrapper).toMatchSnapshot();
    });
});
