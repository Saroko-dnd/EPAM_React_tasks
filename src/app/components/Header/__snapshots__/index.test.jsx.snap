// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render an Header should render Header correctly 1`] = `
<div
  className="site-header d-flex align-items-center flex-column"
>
  <div
    className="d-flex align-items-center"
  >
    <span>
      <NewsIcon
        className="site-header-icon"
        height={150}
        width={150}
      />
    </span>
    <h1>
      News from all over the world
    </h1>
    <button
      className="nav-menu-button d-block d-lg-none"
      onClick={[Function]}
    >
      <span
        className="fa fa-bars"
      />
    </button>
  </div>
  <SlideDown
    className="animation-lg-disabled"
  >
    <NavigationPanel
      className="site-header-navigation d-lg-flex d-none"
      navLinksCallback={[Function]}
      navigationLinks={
        Array [
          Object {
            "destination": "/",
            "id": "nav-link-1",
            "isActive": [Function],
            "title": "Top news",
          },
          Object {
            "destination": "/details",
            "id": "nav-link-5",
            "isActive": null,
            "title": "Detailed news",
          },
          Object {
            "destination": "/second",
            "id": "nav-link-3",
            "isActive": null,
            "title": "Super news",
          },
          Object {
            "destination": "/third",
            "id": "nav-link-4",
            "isActive": null,
            "title": "Mega news",
          },
        ]
      }
    />
  </SlideDown>
</div>
`;
